crates/nu-cli/src/repl.rs

```diff
@@ -203,10 +201,14 @@ pub fn evaluate_repl(
             .with_partial_completions(config.partial_completions)
             .with_ansi_colors(config.use_ansi_coloring);
 
+        let style_computer = StyleComputer::from_config(engine_state, stack);
+
         line_editor = if config.use_ansi_coloring {
-            line_editor.with_hinter(Box::new(
-                DefaultHinter::default().with_style(color_hm["hints"]),
-            ))
+            line_editor.with_hinter(Box::new({
+                // As of Nov 2022, "hints" color_config closures only get `null` passed in.
+                let style = style_computer.compute("hints", &Value::nothing(Span::unknown()));
```

Comment:
What exactly does this mean? Are you saying `hints` won't be colored?

Reply:
What this means is, suppose you write a closure like `hints: {|e| somecommand $e }`. What should the value of $e be? It's currently just `null` because I couldn't think of a good value for it, especially since the style retrieved here gets reused for all hints. (Note that I couldn't do this for search results (in find.rs etc.) because that uses the "string" style for some reason.)

Reply:
I'm sorry for being slow but I'm not understanding. I was thinking that this line meant that it's changing the default color from gray to no color. Perhaps "nothing" means just fall back to the color specified in the config?

I also thought the value of e in your closure meant that it was the current hint. so would this work `hints: {|e| $e | str upcase }` to make all hints the configured color but all uppercase?

Reply:
So, like, in the default config.nu, there's this line:
```
    hints: dark_gray
```
if you edit your config.nu and change it to
```
hints: {|e| somecommand $e }
```
then it will use whatever the output of `somecommand null` is, instead of dark_gray. If `somecommand null` returns an erroneous non-color value, then the current error handling is to just use `Style::default()`. The "

>I also thought the value of e in your closure meant that it was the current hint. so would this work hints: {|e| $e | str upcase } to make all hints the configured color but all uppercase?

The closures defined in this PR *only return colors that already were accepted by `color_config`*.
Here's all permitted types for color_config entries:

* Color name string
* { fg, bg, attr } record
* Closure -> Color name string
* Closure -> { fg, bg, attr } record

>I also thought the value of e in your closure meant that it was the current hint.

The PR currently doesn't do this because I'm concerned about the perf of running the closure every single time the hint is requested. So instead, all hints use one color. (This also applies to search results, as stated earlier).